src/ansys/fluent/core/quantity.py

```diff
@@ -0,0 +1,601 @@
+        temp_quantity.value = (
+            curr_unit_obj.si_factor / to_unit_obj.si_factor
+        ) * self.value
+
+        return temp_quantity
+
+    def _get_si_unit(self, other, func):
+        curr_dim = self.get_dimensions_list()
+        other_dim = other.get_dimensions_list()
+        temp_dim = [func(curr_dim[i], other_dim[i]) for i in range(len(other_dim))]
+        temp_unit = get_si_unit_from_dim(temp_dim)
+        return temp_unit
+
+    def __pow__(self, exponent):
+        new_dims = list(
+            map(lambda x: x * exponent if x != 0 else x, self.get_dimensions_list())
+        )
+        new_si_unit = get_si_unit_from_dim(new_dims)
+        new_si_value = pow(self._si_value, exponent)
+        return Quantity(new_si_value, new_si_unit)
+
+    def __str__(self):
+        return f'({self.value}, "{self.unit}")'
+
+    def __repr__(self):
+        return f'Quantity ({self.value}, "{self.unit}")'
+
+    def __mul__(self, other):
+        if isinstance(other, Quantity):
+            temp_value = self._si_value * other._si_value
+            temp_unit = self._get_si_unit(other, lambda x, y: x + y)
+            return Quantity(temp_value, temp_unit)
+        elif isinstance(other, int) or isinstance(other, float):
+            return Quantity(self._si_value * other, self._si_unit)
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        if isinstance(other, Quantity):
+            temp_value = self._si_value / other._si_value
+            temp_unit = self._get_si_unit(other, lambda x, y: x - y)
+            return Quantity(temp_value, temp_unit)
+        elif isinstance(other, int) or isinstance(other, float):
+            temp = Quantity(self._si_value / other, self._si_unit)
+            return temp
+
+    def __add__(self, other):
+        if isinstance(other, Quantity) and (
+            self.get_dimensions_list() == other.get_dimensions_list()
+        ):
+            temp_value = self._si_value + other._si_value
+        elif self.is_dimension_less() and (
+            isinstance(other, int) or isinstance(other, float)
+        ):
+            temp_value = self._si_value + other
+        else:
+            raise ValueError("Incompatible dimensions.")
+        return Quantity(temp_value, self._si_unit)
+
+    def __radd__(self, other):
+        return self.__add__(other)
+
+    def __sub__(self, other):
+        if isinstance(other, Quantity) and (
+            self.get_dimensions_list() == other.get_dimensions_list()
+        ):
+            temp_value = self._si_value - other._si_value
+        elif self.is_dimension_less() and (
+            isinstance(other, int) or isinstance(other, float)
+        ):
+            temp_value = self._si_value - other
+        else:
+            raise ValueError("Incompatible dimensions.")
+        return Quantity(temp_value, self._si_unit)
+
+    def __rsub__(self, other):
+        return self.__sub__(other)
```

Comment:
This is not correct. You need to do `return Quantity(other) - self`

Reply:
`return Quantity(other, "") - self` causes recursion error.

Reply:
@hpohekar can you identify a relevant failing test with the current code and make the necessary changes to pass the test.

Reply:
@h-krishnan has helped me to resolve this error. It's resolved now. Thank you.